aspnetcore/security/authorization/secure-data/samples/final15/Authorization/ContactRoleAuthorizationHandler.cs

```diff
@@ -16,7 +16,7 @@ public class ContactRoleAuthorizationHandler
                 return Task.FromResult(0);
             }
 
-            if (context.User.IsInRole(requirement.Name))
+            if (context.User.IsInRole(Constants.OperationRolePrefix + requirement.Name))
```

Comment:
I wouldn't link them. I think yes, ContactAdministrators as a role name works, and then the operations map to individual permissions, such as edit and delete. When I demo this I end up having a map of roles to operations/permissions, and end up with two handlers for permissions, one where an administrator can do everything for a permission because they're administrators, and then one handler where the owner of the resource has permission to do an operation.